### ISSUE TITLE
[Bug] + [Feature]: Json Array / postgres text[] type handling

### ISSUE THREAD

**Issue by thebadking** [appsmithorg/appsmith#30077]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

<img width="374" alt="image" src="https://github.com/appsmithorg/appsmith/assets/53491595/06a06981-09d1-49fa-bc22-356fa1593314">

I have this Array of string in postgres that represents multiple choices, they can't be an enum since they are different for every entry (the overlap is insignificant), and since over a year even in other projects I've dealt with this type as a Multiline Input type on appsmith and forced people to write the arrays as such (as on the picture above) and then they need to be parsed as such with a helper function to be inserted into postgres:

const data = JSON.parse(jsonInput);
const pgArray = `{${data.map(item => `"${item}"`).join(", ")}}`;

Field type Array makes a mess of it as it creates objects with sequencial ints as keys and this is not what we need on an array
and to be honest, if I am not mistaken there is a bug there at the moment because selecting it does not allow for inputting any values:
<img width="369" alt="image" src="https://github.com/appsmithorg/appsmith/assets/53491595/79b4bc2a-2580-44f2-b3df-73cf58fcad2b">
you can see that it adds the sequencial keys and the input field for the value cant be clicked for some reason

What we want is an Array type that resembles your Multiselect Type:
<img width="705" alt="image" src="https://github.com/appsmithorg/appsmith/assets/53491595/fb8062cc-a11d-4bbb-bb97-67cf1da89eb6">

But in this case the user would be able to add items to the array dynamically not from a code defined list, sort of what you have on "Tag" interfaces

### Why should this be worked on?

Because it is a very common use of data types and a huge hassle at the moment on both projects that I have going with Appsmith and I can imagine it affects more people

**Comment by Nikhil-Nandagopal** [comment]
@thebadking to understand this correctly, you are looking to automatically handle text arrays so that when they are inserted, appsmith converts the javascript array into { "a", "b", "c" }
Did I get that right?

**Comment by thebadking** [comment]
Yes, back and forward between the Postgres Array and appsmith array, because it requires conversion to insert to the DB
